firmware/Core/Src/Controller.c

```diff
@@ -103,9 +103,43 @@ static Controller_State_t LogData_State(void)
     return CTRL_DO_MATH;
 }
 
+float savedTemps[];
+float loopCounter;
```

Comment:
We should not have variables floating around like this. I suggest you make these static variables inside your state. that way they will preserve their values from call to call while still being local to your function.